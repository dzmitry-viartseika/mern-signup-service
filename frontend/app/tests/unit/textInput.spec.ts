import textInput from '@/components/Elements/TextInput.vue';
import { shallowMount } from '@vue/test-utils';

interface ITextInputProps {
  value: string;
  labelText?: string;
  placeholderText?: string;
  inputType?: string;
  disabled?: boolean;
}

// @Prop({ type: String, required: true })
// value: string;
//
// @Prop({ type: String, required: false })
// placeholderText: string;
//
// @Prop({ type: String, default: 'text' })
// inputType: string;
//
// @Prop({ type: Boolean, default: false })
// disabled: boolean;

describe('textInput component', () => {
  let wrapper: any;

  const createComponent = (props: ITextInputProps) => {
    wrapper = shallowMount(textInput, {
      propsData: props,
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  it('label transfer check', () => {
    createComponent({
      labelText: 'LabelText',
      value: 'testValue',
    });
    expect(wrapper.text()).toContain('LabelText');
  });

  it('check disabled button', () => {
    createComponent({
      disabled: true,
      value: 'testValue',
    });
    const button = wrapper.find('[data-test="input"]');
    expect(button.attributes().disabled).toEqual('disabled');
  });

  // it('renders textInput correctly', () => {
  //   expect(wrapper.element).toMatchSnapshot()
  // })

  // it('placeholder text transfer check', () => {
  //   createComponent({
  //     placeholderText: 'placeholderText',
  //     value: 'testValue'
  //   });
  //   expect(wrapper.text()).toContain('placeholderText');
  // });
});
