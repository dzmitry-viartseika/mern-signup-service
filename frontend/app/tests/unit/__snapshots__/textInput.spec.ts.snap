// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textInput component renders textInput correctly 1`] = `
<div
  class="text-field"
>
  <label
    class="text-field__label"
  >
    
    
  
  </label>
   
  <div
    class="text-field-wrapper"
  >
    <input
      class="text-field__input"
      disabled="disabled"
      type="text"
    />
  </div>
</div>
`;
